internal/rbd/replicationcontrollerserver.go

```diff
@@ -692,3 +697,25 @@ func resyncRequired(localStatus librbd.SiteMirrorImageStatus) bool {
 
 	return false
 }
+
+// repairResyncedImageID updates the existing image ID with new one.
+func repairResyncedImageID(ctx context.Context, rv *rbdVolume, ready bool) error {
+	// During resync operation the local image will get deleted and a new
+	// image is recreated by the rbd mirroring. The new image will have a
+	// new image ID. Once resync is completed update the image ID in the OMAP
+	// to get the image removed from the trash during DeleteVolume.
+
+	// if the image is not completely resynced skip repairing image ID.
+	if !ready {
+		return nil
+	}
+	j, err := volJournal.Connect(rv.Monitors, rv.RadosNamespace, rv.conn.Creds)
+	if err != nil {
+		return err
+	}
+	defer j.Destroy()
+	// reset the image ID which is stored in the existing OMAP
+	rv.ImageID = ""
```

Comment:
Can this be done inside the `repairImageID()` function? Having a requirement like this before calling `repairImageID()` seems difficult to remember, and is easy to break.

Reply:
This is only required for a resync cases, we already have a check-in `repairImageID` as below, doing it explicitly not to disturb the existing code.

```go
func (rv *rbdVolume) repairImageID(ctx context.Context, j *journal.Connection) error {
	if rv.ImageID != "" {
		return nil
	}
```

Reply:
`repairImageID()` is currently called in a single place, you add a 2nd call to it now. Maybe add a `force` bool to the parameters instead?

Reply:
i can do that.

Reply:
i looked at the code. if i add force i need to set `rv.ImageID==""` inside `repairImageID` because `repairImageID` will call `getImageID` and `getImageID` will not fecth the imageID if the rbdVolume.ImageID is set. 

is this fine?

Reply:
fixed now. PTAL